python/lsst/verify/ap/pipeline.py

```diff
@@ -0,0 +1,74 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+
+
+class Pipeline(object):
```

Comment:
Sort of. It's for clarifying exactly what an alternative to `ApPipe` needs to look like (i.e., it must work in terms of a `Dataset` and an output repo, and it must have a `run` method), and dividing things needed by *any* pipeline wrapper from things specific to wrapping `ap_pipe`.

It's a very C++ic way of approaching the problem, but @parejkoj assures me it's not discouraged in Python (though he did suggest some changes if I keep this class).